filebeat/tests/system/test_crawler.py

```diff
@@ -44,7 +44,7 @@ def test_fetched_lines(self):
         # TODO: Find better solution when filebeat did crawl the file
         # Idea: Special flag to filebeat so that filebeat is only doing and
         # crawl and then finishes
-        filebeat.kill_and_wait()
+        assert 0 == filebeat.kill_and_wait()
```

Comment:
I would suggest to make this check directly inside kill_and_wait(). kill_and_wait would have a paramt (check_exit) which is true by default.